std/functional.d

```diff
@@ -260,6 +210,127 @@ unittest
     static assert(!is(typeof(binaryFun!FuncObj)));
 }
 
+/**
+Transforms a string representing an expression into a n-ary function.
+$(UL
+    $(LI The number of parameters is given by `argc`. If no explicit
+        `argc` is given, it is inferred to be `parmNames.length`.)
+    $(LI Explicit names for the function parameters may be given as
+        `parmNames`. If more than `argc` names are supplied, the
+        extras will be ignored.)
+    $(LI The letters (`a` through `z`) are used as default names
+        for up to the first 26 function parameters.)
+    $(LI `args[index]` can always be used to access any parameter by index,
+        regardless of whether it has been given a name.)
+)
+*/
+template stringLambda(alias fun, size_t argc, parmNames...)
+    if (parmNames.length == 0 || allSatisfy!(isSomeString, typeof(parmNames)))
+{
+    static if (is(typeof(fun) : string))
+    {
+        // No need to regenerate the aliases mixin for every ElementTypes permutation.
+        private enum aliases = stringLambdaAliases!(argc, parmNames);
+
+        static if (!fun._ctfeMatchNary(parmNames))
+        {
+            import std.traits, std.typecons, std.typetuple;
+            import std.algorithm, std.conv, std.exception, std.math, std.range, std.string;
+        }
+
+        auto stringLambda(ArgTypes...)(auto ref ArgTypes args)
+            if (ArgTypes.length == argc)
+        {
+            mixin(aliases);
+            return mixin(fun);
+        }
+    }
+    else static if (needOpCallAlias!fun)
+    {
+        // Issue 9906
+        alias stringLambda = fun.opCall;
+    }
+    else
+    {
+        alias stringLambda = fun;
+    }
+}
+
+/// ditto
+template stringLambda(alias fun, parmNames...)
+    if (allSatisfy!(isSomeString, typeof(parmNames)))
+{
+    alias stringLambda = stringLambda!(fun, parmNames.length, parmNames);
+}
+
+// Generate stringLambda parameter aliases mixin string.
+private template stringLambdaAliases(size_t argc, parmNames...)
+{
+    import std.conv : to;
+    import std.array : appender;
+    import std.algorithm : min;
+
+    // use a letter for each unspecified param, stopping at 'z'
+    enum numLetters = min(argc, 'z' - 'a' + 1);
+
+    auto stringLambdaAliases()
```

Comment:
I threw a pragma msg in there and it only seems to instantiate once for a given set of args.

A simpler example:

    template temp(T...) {
      auto temp() {
        pragma(msg, T);
      }
    }

    unittest {
      temp!int;
      temp!int;
      temp!int;
    }

The nested `temp` only gets evaluated once.